### ISSUE TITLE
Feature request for Cloudwatch metrics collection for all Namespaces

### ISSUE THREAD

**Issue by mohag1412** [influxdata/telegraf#6961]
## Feature Request

Opening a feature request kicks off a discussion.

**Comment by mohag1412** [comment]
Hi,

I was looking for a possibility to collect all CloudWatch metrics from all the AWS namespaces.
Currently Telegraf supports cloudwatch metrics collection with limited namesapces.

We have a requirement, where there are around 30+ namespaces and metrics needs to be collected from all of them. Also, the Cloudwatch Input plugin supports one region per input block.

In a scenario, where I need cloud watch metrics from all AWS regions and for all namespaces we need to add 30*10=300 Input blocks.

Can adding multi-region and multi namespaces can be considered in single input plugin?

[[inputs.cloudwatch]]
 region = "us-east-1, us-west-1, eu-west-1"
 period = "5m"
 access_key = "accesskey"
 secret_key = "secretkey"
 namespace = "AWS/RDS, AWS/EC2,AWS/S3"
 ratelimit = 200

**Comment by danielnelson** [comment]
Thanks for the feature request, but I'm unsure if it will work well.  Each namespace has a different set of metric dimensions, so you wouldn't be able to use it in conjunction with manual dimensions, and supporting it will likely cause confusion setting up the plugin.  To make things worse, each namespace has a different requirement for the `period`, and sometimes even multiple settings.

**Comment by mohag1412** [comment]
Thanks, Daniel for your response. We are using Datadog and all cloudwatch metrics get pushed once Integration is done. Since we are looking for an alternative to Datadog, we are evaluating Telegraf.

Is there a way with Telegraf where we can get most of the CloudWatch metrics pushed along with EC2 tags in a single Input plugin. 

With Telegraf Cloudwatch Input plugin, EC2 tags doesn't get populated ( Unlike to Datadog).
Any inputs on how to get EC2 tags to get populated with Telegraf cloudwatch plugin ?

**Comment by mohag1412** [comment]
Also Daniel,  how about considering multi-region in same input plugin?

**Comment by danielnelson** [comment]
I think multi region could be done.

For the instance tags, we don't have a way to collect them yet, it is an entirely different API compared to the Cloudwatch API.  We have an issue for this: https://github.com/influxdata/telegraf/issues/6343

**Comment by mohag1412** [comment]
Thanks Daniel for raising the issue. I would be awaiting for updates on multi-region Cloudwatch metrics enablement .

Regards
Greeta

**Comment by mohag1412** [comment]
Daniel/Team,
Can getting CloudWatch metrics from multiple AWS accounts be taken into consideration?

**Comment by danielnelson** [comment]
I think for that we will want to stick with multiple plugins, I feel like it would add too much complexity to the configuration due to how many options are permission related.

Telegraf isn't optimized for small configuration file size, so I often recommend those with more complex needs generate their config from a set of template files.

**Comment by reimda** [comment]
There are three features discussed in this issue: support for multiple namespaces, multiple accounts, and multiple regions. Namespaces and accounts are handled by using multiple plugins. Multiple regions are still not supported in a single plugin instance.

I'm closing this issue due to age and because it requests three different features. If there is still a need for multiple regions in one plugin instance, please open a new issue to address it. Thanks!
